core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -108,13 +115,6 @@ final class ShuffleBlockFetcherIterator(
 
   private[this] val startTimeNs = System.nanoTime()
 
-  /** Local blocks to fetch, excluding zero-sized blocks. */
-  private[this] val localBlocks = scala.collection.mutable.LinkedHashSet[(BlockId, Int)]()
-
-  /** Host local blockIds to fetch by executors, excluding zero-sized blocks. */
-  private[this] val hostLocalBlocksByExecutor =
-    LinkedHashMap[BlockManagerId, Seq[(BlockId, Long, Int)]]()
-
```

Comment:
Note to reviewers: Both of these are created locally in the `initialize` and `fetchFallbackBlocks`. With push-based shuffle, `partitionBlocksByFetchMode` can be called multiple times because failure to fetch merged shuffle blocks/chunks (fallback) finds original blocks that made up the failed merged block/chunk and then we need to create new FetchRequests from these.